torch/_inductor/runtime/runtime_utils.py

```diff
@@ -81,35 +80,62 @@ def do_bench(fn, fn_args, fn_kwargs, **kwargs):
         return do_bench_gpu(lambda: fn(*fn_args, **fn_kwargs), **kwargs)
 
 
-def do_bench_gpu(*args, **kwargs):
+def do_bench_gpu(fn, warmup=25, rep=100, fast_flush=True):
     @functools.lru_cache(None)
-    def load_triton():
-        try:
-            # NB: Lazily load triton, as importing triton is slow
-            # see https://github.com/openai/triton/issues/1599
-            from triton.testing import do_bench as triton_do_bench
-        except ImportError as exc:
-            raise NotImplementedError("requires Triton") from exc
-
-        # triton PR https://github.com/openai/triton/pull/1513 change the
-        # quantile fields name from 'percentiles' to 'quantiles'
-        # and change the default value from (0.5, 0.2, 0.8) to None.
-        # This may break inductor since a caller expects a tuple may get a item.
-        #
-        # Add a wrapper to maintain the same behavior for inductor.
-        # Maybe we should have own implementation of this function?
-        return triton_do_bench, (
-            "quantiles"
-            if inspect.signature(triton_do_bench).parameters.get("quantiles")
-            is not None
-            else "percentiles"
+    def get_cache_size():
+        device = torch.cuda.current_device()
+        properties = torch.cuda.get_device_properties(device)
+        return properties.l2CacheSize
```

Comment:
>  (A problem with flushing exactly the L2 cache size is that you don't actually know that this clears the cache. For example H100 has two separate L2s, and I don't know how they work. But again, touching 4MB vs 256MB should not be a substantial difference, certainly not >1ms on H100, so if you're observing that, it seems like something is wrong.)

Should we double on h100 ?

Reply:
As far as I know, H100 has a single L2 cache; I've done some research and nothing came up mentioning two L2 caches, and experimentally 50MB flushes showed equivalent performance (whereas 49MB did not produce the correct results). I'm inclined to believe this was a typo on their part, probably meant B100. In that case, I have no idea how Nvidia will handle this. Should their CUDA device properties return the total L2 cache size? Probably. I won't know until we get B100s to test. In the meantime I don't think this should be a big issue due to the incredibly limited quantity of B100s out in the wild. My opinion here is that we just land now and check when we have access later, but I could see why this isn't amazing either.